src/core/geometry/qgsregularpolygon.h

```diff
@@ -0,0 +1,213 @@
+/***************************************************************************
+                         qgsregularpolygon.h
+                         --------------
+    begin                : May 2017
+    copyright            : (C) 2017 by Loîc Bartoletti
+    email                : lbartoletti at tuxfamily dot org
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSREGULARPOLYGON_H
+#define QGSREGULARPOLYGON_H
+
+#include <QString>
+
+#include "qgis_core.h"
+#include "qgspointv2.h"
+#include "qgspolygon.h"
+#include "qgslinestring.h"
+#include "qgscircle.h"
+#include "qgstriangle.h"
+
+enum ConstructionOption
+{
+  inscribedCircle,
+  circumscribedCircle
+};
+
+/** \ingroup core
+ * \class QgsRegularPolygon
+ * \brief Regular Polygon geometry type.
```

Comment:
Can you add a bit more detail here? Something like the wikipedia definition: "A regular polygon is a polygon that is equiangular (all angles are equal in measure) and equilateral (all sides have the same length). Regular polygons may be convex or star and include shapes like hexagons and pentagons."